app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -74,45 +75,33 @@ class EnterMovementReferenceNumberController @Inject() (
       }
     }
 
-  protected def renderMrnPage(answers: MovementReferenceNumberAnswer, isAmend: Boolean)(implicit
+  protected def renderMrnPage(answers: Option[MovementReferenceNumber], isAmend: Boolean)(implicit
     request: RequestWithSessionData[_]
   ): Result =
-    answers.fold(
-      ifIncomplete =>
-        ifIncomplete.movementReferenceNumber match {
-          case Some(movementReferenceNumber) =>
-            Ok(
-              enterMovementReferenceNumberPage(
-                EnterMovementReferenceNumberController.movementReferenceNumberForm.fill(
-                  MovementReferenceNumber(movementReferenceNumber)
-                ),
-                isAmend = isAmend
-              )
-            )
-          case None                          =>
-            Ok(
-              enterMovementReferenceNumberPage(
-                EnterMovementReferenceNumberController.movementReferenceNumberForm,
-                isAmend = isAmend
-              )
-            )
-        },
-      ifComplete =>
+    answers match {
+      case Some(movementReferenceNumber) =>
         Ok(
           enterMovementReferenceNumberPage(
             EnterMovementReferenceNumberController.movementReferenceNumberForm.fill(
-              MovementReferenceNumber(ifComplete.movementReferenceNumber)
+              movementReferenceNumber
```

Comment:
replaced by the match on the optional MovementReferenceNumberAnswer (could just as easily be a fold). The inner pattern match is gone completely because if an answer was stored then it would never not be there so it didn't need to be optional.